server/chat-plugins/blackjack.js

```diff
@@ -0,0 +1,641 @@
+				points += 10;
+			} else if (card === 'A') {
+				points += 11;
+				aceCount++;
+			}
+		}
+
+		// At first, we value aces as 11, however, we will change their value
+		// to be 1 if having them as 11 would cause an unnecessary bust. We will
+		// do this by subtracting 10 for each ace that would otherwise cause a bust.
+		while (points > 21 && aceCount > 0) {
+			points -= 10;
+			aceCount--;
+		}
+
+		return points;
+	}
+	next() {
+		this.clearAllTimers();
+		if (Object.keys(this.playerTable)[Object.keys(this.playerTable).length - 1] === this.curUser && this.playerTable[this.curUser].status !== 'playing') {
+			if (this.dealer.points < 17) {
+				this.giveCard('dealer');
+			} else if (this.dealer.points >= 17) {
+				let cards = '';
+				for (let card of this.dealer.cards) cards += `[${card}] `;
+				let turnLine = `<br /><strong>${this.dealer.name}</strong> stands with ${cards} (${this.dealer.points})`;
+				this.turnLog += turnLine;
+				this.display(turnLine);
+				this.end();
+			}
+			return;
+		}
+		if (this.playerTable[this.curUser].status !== 'playing') {
+			let number = 0;
+			while (this.playerTable[Object.keys(this.playerTable)[number]].status !== 'playing') number++;
+			this.curUser = Object.keys(this.playerTable)[number];
+		}
+		let output = `|uhtml${this.playerTable[this.curUser].selfUhtml}|user-blackjack-${this.blackjackNumber}|<div class="infobox">`;
+		output += `It's your turn to move, ${this.playerTable[this.curUser].name}<br />`;
+		for (let card of this.playerTable[this.curUser].cards) {
+			output += this.generateCard(card);
+		}
+		output += `<br />Score: ${this.playerTable[this.curUser].points}${(this.playerTable[this.curUser].points === 21 ? ` (you have blackjack!)` : ``)}`;
+		output += `<br /><button class="button" name="send" value="/blackjack hit" title="Hit (get another card)">Hit</button> | <button class="button" name="send" value="/blackjack stand" title="Stand (just keep these cards)">Stand</button>`;
+
+		this.playerTable[this.curUser].sendRoom(`|notify|Blackjack (${this.room.title})|It's your turn to play!`);
+		this.playerTable[this.curUser].sendRoom(output);
+		this.playerTable[this.curUser].playScreen = output;
+
+		this.createTimer(this.playerTable[this.curUser]);
+	}
+}
+
+class BlackjackPlayer extends Rooms.RoomGamePlayer {
+	constructor(user, game) {
+		super(user, game);
+		this.game = game;
+
+		this.cards = [];
+		this.points = 0;
+		this.slide = 0;
+		this.status = 'playing';
+
+		this.selfUhtml = '';
+		this.gameLog = '';
+		this.playScreen = '';
+		this.timerTicksLeft = this.game.turnTimeoutMinutes * 60; // to get into seconds-format
+	}
+	resetTimerTicks() {
+		this.timerTicksLeft = this.game.turnTimeoutMinutes * 60;
+	}
+}
+
+class BlackjackDealer {
+	constructor() {
+		this.cards = [];
+		this.points = 0;
+		this.name = 'The Dealer';
+	}
+}
+
+class BlackjackDeck {
+	constructor() {
+		this.deck = [
+			'A♥', 'A♦', 'A♣', 'A♠', '2♥', '2♦', '2♣', '2♠', '3♥', '3♦', '3♣',
```

Comment:
Not critical if youre sure you havent typoed, but alternatively:

```ts
const ranks = 'A2345678910JQK'.split('');
const suits = '♥♦♣♠'.split('');
this.deck = ranks.flatMap(rank => suits.map(suit => `${rank}${suit}`));
```

Reply:
Yeah, I'm sure I haven't typo'd